dChatFilter/dChatFilter.cpp

```diff
@@ -86,46 +95,56 @@ bool dChatFilter::ReadWordlistDCF(const std::string& filepath) {
 	return false;
 }
 
-void dChatFilter::ExportWordlistToDCF(const std::string& filepath) {
+void dChatFilter::ExportWordlistToDCF(const std::string& filepath, bool whiteList) {
 	std::ofstream file(filepath, std::ios::binary | std::ios_base::out);
 	if (file) {
 		BinaryIO::BinaryWrite(file, uint32_t(dChatFilterDCF::header));
 		BinaryIO::BinaryWrite(file, uint32_t(dChatFilterDCF::formatVersion));
-		BinaryIO::BinaryWrite(file, size_t(m_Words.size()));
+		BinaryIO::BinaryWrite(file, size_t(whiteList ? m_YesYesWords.size() : m_NoNoWords.size()));
 
-		for (size_t word : m_Words) {
+		for (size_t word : whiteList ? m_YesYesWords : m_NoNoWords) {
 			BinaryIO::BinaryWrite(file, word);
 		}
 
 		file.close();
 	}
 }
 
-bool dChatFilter::IsSentenceOkay(const std::string& message, int gmLevel) {
-	if (gmLevel > GAME_MASTER_LEVEL_FORUM_MODERATOR) return true; //If anything but a forum mod, return true.
-	if (message.empty()) return true;
+std::vector<std::string> dChatFilter::IsSentenceOkay(const std::string& message, int gmLevel, bool whiteList) {
+	if (gmLevel > GAME_MASTER_LEVEL_FORUM_MODERATOR) return { }; //If anything but a forum mod, return true.
+	if (message.empty()) return { };
+	if (!whiteList && m_NoNoWords.empty()) return { "" };
```

Comment:
Why is this one an empty string vs the other two above?

Reply:
Because that is disallowing chat if someone removes the whitelist